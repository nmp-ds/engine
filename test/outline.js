import { setup } from "./_helpers.js";
import { describe, expect, test } from "vitest";
import { lineWidth } from '#theme';

setup();

describe("outline", () => {
  test("style", async ({ uno }) => {

    const classes = [
      "outline-none",
      "outline",
      "outline-dashed",
      "outline-dotted",
      "outline-double",
    ];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });

  test("width", async ({ uno }) => {
    const classes = Object.keys(lineWidth).map(width => [`outline-${width}`]).flat();
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();

  });
  test("offset", async ({ uno }) => {
    const classes = Object.keys(lineWidth).map(width => [`outline-offset-${width}`]).flat();
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
});