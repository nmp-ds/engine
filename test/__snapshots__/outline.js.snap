// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`outline > offset 1`] = `
"/* layer: default */
.outline-offset-0{outline-offset:0;}
.outline-offset-1{outline-offset:1px;}
.outline-offset-2{outline-offset:2px;}
.outline-offset-4{outline-offset:4px;}
.outline-offset-8{outline-offset:8px;}"
`;

exports[`outline > style 1`] = `
"/* layer: default */
.outline-none{outline:2px solid transparent;outline-offset:2px;}
.outline{outline-style:solid;}
.outline-dashed{outline-style:dashed;}
.outline-dotted{outline-style:dotted;}
.outline-double{outline-style:double;}"
`;

exports[`outline > width 1`] = `
"/* layer: default */
.outline-0{outline-width:0;}
.outline-1{outline-width:1px;}
.outline-2{outline-width:2px;}
.outline-4{outline-width:4px;}
.outline-8{outline-width:8px;}"
`;
